Gamble/collect

hardik281199 opened the pull request.

hardik281199 committed:
some changes

hardik281199 committed:
create : seeder &&  modified :all static get in slot-game main function AND add promise

hardik281199 committed:
response dispatcher && json.message

hardik281199 committed:
change name of disppatcher && commenting

hardik281199 committed:
change name logout

hardik281199 committed:
generateViewZone

hardik281199 committed:
static row&colume store in data base and commenting seeder

hardik281199 committed:
add: function modularization

hardik281199 committed:
function modularization and conting sactter

hardik281199 committed:
Expanding Wild : creat expandingWild function

hardik281199 committed:
count Wild Care and replace other symbole

hardik281199 committed:
wild/multiplier

hardik281199 committed:
wild/multiplier

hardik281199 committed:
solve error : free game responce ,commenting

hardik281199 committed:
solve error : free game responce ,commenting

hardik281199 committed:
gamble : gamble counting ,response gamble function and routing and error validation
collect : collect win amount and add wallet
spin : validation spin winAmount collect or not
userINFO: set response

hardik281199 committed:
gamble : gamble counting ,response gamble function and routing and error validation
collect : collect win amount and add wallet
spin : validation spin winAmount collect or not
userINFO: set response

hardik281199 committed:
limit: limit in Gamble WinAmount
moster : set DEVIL symbole

hardik281199 committed:
limit: limit in Gamble WinAmount
moster : set DEVIL symbole

alishascaletech commented on src/utils/gamehelper.js:
```diff
@@ -38,7 +38,7 @@ class GameHelper{
         const row =sta.viewZone.rows;
         const colume = sta.viewZone.columns;           
-        const randomNumber = this.randomInt(0,9);
+        const randomNumber = this.randomInt(0,10);
```
It should be `this.randomInt(0,reelArray.length);`

alishascaletech commented on src/utils/gamehelper.js:
```diff
@@ -137,7 +151,24 @@ class GameHelper{
                
         }
-        return {viewZone : viewZone ,generatedArray : newGenerateArray , expandingWild : expandingWild }  
+        return {viewZone : viewZone ,generatedArray : newGenerateArray , expandingWild : expandingWild ,wildMultipliar : wildMultipliar,wildMultipliarArray : wildMultipliarArray , checkDevil : checkDevil }  
```
Use short hand property here

alishascaletech commented on src/utils/gamehelper.js:
```diff
@@ -137,7 +151,24 @@ class GameHelper{
+        let wildMultipliar ={
+            row : row,
+            colume : reel,
+            Multipliar:random
```
`Multiplar` should be `multiplier`
`colume` should be `column`

alishascaletech commented on src/utils/gamehelper.js:
```diff
@@ -167,7 +198,7 @@ class GameHelper{
      * @returns 
      */
-    checkPayline=(payarray,matrixReelXCol,content,Pay) =>{
+    checkPayline=(payarray,matrixReelXCol,content,Pay,checkDevil) =>{
```
`Pay` should be `pay` as per camelCasing naming convention

alishascaletech commented on src/utils/gamehelper.js:
```diff
@@ -236,6 +270,16 @@ class GameHelper{
+     * @returns wild multipair
+     */
+    randomWildMul(){
+        let wildMult = [2,4,6]
```
Set `wildMult` in database game object. It should not be declared static here.

alishascaletech commented on src/utils/gamehelper.js:
```diff
@@ -236,6 +270,16 @@ class GameHelper{
+     */
+    randomWildMul(){
+        let wildMult = [2,4,6]
+        const element = Math.floor(Math.random() * ((2-0)+ 1) + 0);
```
` Math.floor(Math.random() * ((2-0)+ 1) + 0);`
it can be written as `Math.floor(Math.random() * wildMult.length());`

alishascaletech commented on src/utils/gamehelper.js:
```diff
@@ -296,6 +346,65 @@ class GameHelper{
+     * @returns card
+     */
+    randomGambleCard = (low,high)=>{
+        let gambleCard = ["black","red"];
```
["black","red"];
should be defined & supllied from `constants` file

alishascaletech commented on src/utils/gamehelper.js:
```diff
@@ -296,6 +346,65 @@ class GameHelper{
+            gamble_history = [];
+            winInSpin = 0;
+        }
+        return {gambleWin : gambleWin,winInSpin : winInSpin , gamblecounter : gamblecounter,gamble_history : gamble_history }
```
Use short hand property

alishascaletech commented on src/utils/gamehelper.js:
```diff
@@ -296,6 +346,65 @@ class GameHelper{
+        }
+        winInSpin = 0;
+        wildMultipliar = 0;
+        return { wallet : wallet ,winInSpin : winInSpin , gamblecounter : gamblecounter , gambleWin : gambleWin , wildMultipliar : wildMultipliar}
```
Use short hand property

alishascaletech commented on src/utils/userhelper.js:
```diff
@@ -0,0 +1,29 @@
+     * @param {res} res 
+     * @returns account detail
+     */
+    gameFunction(req,res) {
```
rename it to `buildUserData()` or something else which convey the purpose of function

alishascaletech commented on src/utils/userhelper.js:
```diff
@@ -0,0 +1,29 @@
+const Bcrypt = require('bcrypt');
+class GameFunction{
```
rename it to `UserHelper`

alishascaletech commented on src/utils/userhelper.js:
```diff
@@ -0,0 +1,29 @@
+        return account;
+    }
+}
+const game = new GameFunction();
```
object name should be `user` not `game`

alishascaletech reviewed (changes requested):
Refer to comments & make necessary changes

hardik281199 committed:
modified: Use short hand property && Camel-case variables && Add config contants file && devil counting error solve

hardik281199 committed:
modified: gamehelper ->randomInt
		->Use short hand property
		-> change name
		->camelCasing
		->all static data stroe in dataBase
	userhelper change name
	add api ==gameConfig

hardik281199 committed:
some change in Pay to pay

hardik281199 committed:
Merge branch 'gamble/collect' of https://github.com/hardik281199/slot-game into gamble/collect

hardik281199 committed:
validate gameConfig using middleware
